accomodate more general tidy format and datasets

BramVDBe opened the pull request:
I had some issues with using the tidyformat in QurvE and discovered that description or condition should not be coded as a factor, multiple blanks should have different replicate numbers and single repeat data (so with only one single sample) cannot be processed. 

Therefore, I included some changes in the functions tidy_to_custom, read_data and growth.workflow

BramVDBe committed:
Update data_parsers.R

handle factorial description data by converting with as.character in tidy_to_cutstom

BramVDBe committed:
Update data_parsers.R

initialize colnames of dat.mat matrix to rename first 3

BramVDBe committed:
Update data_parsers.R

conditional check in tidy_to_custom to accomodate multiple blanks reps with 'same' replicate numbers

BramVDBe committed:
Update growth_workflows.R

allow single repeat single condition data to be fitted

BramVDBe committed:
Update data_parsers.R

some small changes

BramVDBe committed:
Update growth_workflows.R

typo

BramVDBe committed:
Update data_parsers.R

copied working read_data and tidy_to_custom from r script...

BramVDBe committed:
Update data_parsers.R

NicWir reviewed (commented):
If there exist multiple occurrences of the same combination of "Description", "Replicate", and "Concentration, during conversion into long format, the values will be extracted into a vector in the form:
[A1, B1, A2, B2, A3, B3, ...] (with t1, t2, t3, ... time values)

The suggested fix with determining nreps and looping through these integer values would extract:
[A1, [A1, B1, A2, B2, A3, B3, ..., B(0.5*tmax)] and [A(0.5tmax+1),  ..., B(tmax)]  and thus the time values would not match the measurements

BramVDBe committed:
Update data_parsers.R

correct selection of time points/values in case non unique datasets (blanks or otherwise) are included

NicWir reviewed (commented):
Thanks! I implemented these changes

NicWir reviewed (commented):
Thanks a lot for this fix! I implemented it (will not merge though to avoid conflicts)

NicWir reviewed (commented):
implemented this. The if-chunks need a {}, though. Placing a conditional statement after if() without curly brackets only works for a single line, not multiple lines

NicWir reviewed (commented):
thanks also for this suggested change to avoid ugly matrix headers. Implemented!

NicWir reviewed (commented):
It is very hard to judge if there are any other changes in here than were part of the previous commits. If so, please let me know

NicWir commented:
I implemented some of the suggested changes with some modifications and implemented an alternative fix for the described issue. Thank you for your contribution!
